### PR TITLE
Edit hometown

### PR DESCRIPTION
**Pull request by JohnPMurray**
Edit works. Tests pending

Size M:
As an admin I want to be able to change a student's stored hometown so that it can be correct in the case of a spelling error or mistake.

[https://trello.com/c/Vfh9yK8S/80-admin-edit-student-hometown](https://trello.com/c/Vfh9yK8S/80-admin-edit-student-hometown)

### CODE REVIEW

**Review by johnbyrne727rit — COMMENTED** [review]
Looks good to me so far.

File: backend/schema.js [hunk 1]
```diff
@@ -268,7 +269,7 @@ type Mutation {
     createJudge(input: UserInput!): User
     createAdmin(input: UserInput!): User
     updatePermissions(input: PermissionInput!): User
-    updateUser(id: ID!, input: UserInput!): User
```
Comment (JohnPMurray) [inline backend/schema.js:271]: I don't think update user gets called anywhere else but I'll have to double check to make sure I didn't break anything

File: frontend/src/Admin/components/EditHometown.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            <Row>
+              <Col>
+                <Field
+                    type='text'
+                    id='hometown'
+                    name='hometown'
+                    className='form-control'
+                  />
+              </Col> 
+              <Button
```
Comment (JohnPMurray) [inline frontend/src/Admin/components/EditHometown.js]: Will need to do some aesthetic updates on the done button it looks kinda weird

File: frontend/src/Admin/components/StudentsTable.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react'
```
Comment (JohnPMurray) [inline frontend/src/Admin/components/StudentsTable.js:1]: This file is copied from the UserTable file with the hometown field added, which is a form that can be edited, see EditHometown file

File: backend/schema.js (hunk 1 above)
Comment (JohnPMurray) [reply backend/schema.js:271]: tests all pass so this should be fine

File: frontend/src/Admin/components/EditHometown.js (hunk 2 above)
Comment (JohnPMurray) [reply frontend/src/Admin/components/EditHometown.js]: done

**Comment by JohnPMurray** [comment]
Screen Shot 2020-02-03 at 9 47 20 AM
Screen Shot 2020-02-03 at 9 47 03 AM

**Comment by JohnPMurray** [comment]
Finished all the tests and performed some aesthetic updates. I left a couple screen grabs above.

File: frontend/src/Admin/components/EditHometown.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+          firstName: student.firstName,
+          lastName: student.lastName,
+          hometown: student.hometown,
+          isEditing: 'no',
+        }}
+        validationSchema={yup.object().shape({
+          username: yup.string().required('Required'),
+          firstName: yup.string().required('Required'),
+          lastName: yup.string().required('Required'),
+          hometown: yup.string().nullable()
```
Comment (johnbyrne727rit) [inline frontend/src/Admin/components/EditHometown.js:54]: Do we want to let the Admin just delete a hometown? Should they have to give it a value?

Comment (JohnPMurray) [reply frontend/src/Admin/components/EditHometown.js:54]: I don’t see why not, they are an admin

Comment (JohnPMurray) [reply frontend/src/Admin/components/EditHometown.js:54]: It doesn’t break anything if they do. As far as I can tell it just asks the user for their hometown again, which might be the best way to do it

Comment (johnbyrne727rit) [reply frontend/src/Admin/components/EditHometown.js:54]: Fair.

**Review by johnbyrne727rit — APPROVED** [review]
Looks good to me.

**Review by gollb310 — APPROVED** [review]
👍

**Pull request merged**